views/default/elgg/lightbox.js

```diff
@@ -107,7 +105,29 @@ define(function (require) {
 						opts.href = elgg.getSelectorFromUrlFragment(opts.href);
 					}
 
-					lightbox.open(opts);
+					if (opts.photo || opts.inline || opts.iframe || opts.html) {
+						lightbox.open(opts);
+					} else {
+						
+						var href = opts.href
+						opts.href = false;
+						var data = opts.data;
+						opts.data = undefined;
+						
+						// open lightbox without a href so we get a loader
+						lightbox.open(opts);
+						
+						require(['elgg/Ajax'], function(Ajax) {
+							var ajax = new Ajax(false);
+							ajax.path(href, {data: data}).done(function(output) {
+								opts.html = output;
+								lightbox.open(opts);
+								
+								// clear data so next fetch will refresh contents
+								opts.html = undefined;
```

Comment:
100%... if you reopen the lightbox... the content would otherwise still be there (and you would see no loader)